build: move CHANGELOG to public-api approval group

kara opened the pull request:
Previously, the CHANGELOG file was caught under the general
"*" glob, which meant that it fell under the dev-infra
approval group. This does not really make sense since it
has more to do with public-facing changes. As such, this
commit moves the CHANGELOG file specifically to the
public-api approval group.

kara committed:
build: move CHANGELOG to public-api approval group

Previously, the CHANGELOG file was caught under the general
"*" glob, which meant that it fell under the dev-infra
approval group. This does not really make sense since it
has more to do with public-facing changes. As such, this
commit moves the CHANGELOG file specifically to the
public-api approval group.

IgorMinar commented on .pullapprove.yml:
```diff
@@ -1020,6 +1020,7 @@ groups:
       - >
         contains_any_globs(files, [
           'goldens/public-api/**',
+          'CHANGELOG.md',
```
I believe that this change alone is not sufficient. we also need to exclude this file from the `dev-infra` group. otherwise we'll need both the `public-api` group and the `dev-infra` groups to approve any changes to the CHANGELOG.md file.

kara replied:
ohh good point.

IgorMinar reviewed (changes requested):
What if we removed the `'*'` rule from `dev-infra` group and explicitly specified which of the top level files belong to `dev-infra` and which belong elsewhere?

It seems to me that top level `'*'` is a recipe for more confusion in the future. Being explicit will prevent that.

IgorMinar commented:
I can help make the change and push to this branch if you'd like.

kara committed:
fixup! build: move CHANGELOG to public-api approval group

IgorMinar commented:
I also think that we should add you to the `public-api` group even though we don't have staged reviews yet. That way you can help with approving changelog.md and other changes that are straightforward and don't need much discussion.

sidenote: I'd love for us to have a public-api "club" where we discuss public api changes and mentor more people on how to evolve the public api surface in a way that is backwards compatible, ergonomic, consistent, and enables incremental evolution. Would you be interested in that?

IgorMinar reviewed (approved):
lgtm, now. We can either make the remaining changes here or in a follow up PR. You pick.

kara committed:
build: add kara to public-api approvers

This commit adds Kara (me) to the public-api group
so that we can start to divide the public-api code
reviews among more people.

IgorMinar reviewed (approved):
LGTM!

Reviewed-for: global-approvers
